### DISCUSSION TITLE
In-game scaling and font settings

### DISCUSSION THREAD

**Discussion by nielsmh** [OpenTTD/OpenTTD#9965] [Features]
I drew an idea for a new UI for scaling and font settings.

![image](https://user-images.githubusercontent.com/1062071/184479835-6c810787-c374-435c-b936-0e31146de952.png)

There are two new things here that don't exist in the game currently.

First is _Video scaling_, this would be a full-screen scaling implemented in the graphics driver. The graphics driver would be required to ask the viewport and GUI rendering system to render the entire game at a lower resolution than the actual window size/screen resolution, and then the driver scales up the result. This will allow non-integer scaling, although it won't look great. Accelerated graphics drivers could use interpolation for some scaling settings.

The other is _in-game font selection_. The check boxes are called _Prefer pixel font_, since the game will still "fall back" to a scalable font for languages that aren't supported by the pixel font, e.g. CJK and Arabic languages. The "Select" button for fonts should probably open an OS-dependent font selection window, or possibly a game-GUI window that just requests a list of available fonts from the OS.

I just noticed I forgot a setting for _Monospace font_ in this mockup.

The window might also benefit from a kind of preview pane, but that might be difficult to do without taking up as much space as the rest of the window.

The video scaling feature would somewhat clash with resolution setting, and perhaps that should also be added to this window. At the same this it might also be an opportunity to move towards "windowed fullscreen", where the selected resolution for fullscreen is not switched to, but desktop resolution remains as when the game started and then the graphics is just rendered at the requested resolution and scaled to fit.

**Comment by PeterN** [comment]
I have a freeform scaling patch that implements some of this already. Needs a bit of work to bring up to date.
